diff --git a/applications/services/gui/modules/text_input.c b/applications/services/gui/modules/text_input.c
index 50453cf22..f251b777b 100644
--- a/applications/services/gui/modules/text_input.c
+++ b/applications/services/gui/modules/text_input.c
@@ -30,6 +30,8 @@ typedef struct {
     void* validator_callback_context;
     FuriString* validator_text;
     bool validator_message_visible;
+
+    size_t minimum_length;
 } TextInputModel;
 
 static const uint8_t keyboard_origin_x = 1;
@@ -530,6 +532,14 @@ void text_input_set_result_callback(
         true);
 }
 
+void text_input_set_minimum_length(TextInput* text_input, size_t minimum_length) {
+    with_view_model(
+        text_input->view,
+        TextInputModel * model,
+        { model->minimum_length = minimum_length; },
+        true);
+}
+
 void text_input_set_validator(
     TextInput* text_input,
     TextInputValidatorCallback callback,
diff --git a/applications/services/gui/modules/text_input.h b/applications/services/gui/modules/text_input.h
index 218df3141..65207532e 100644
--- a/applications/services/gui/modules/text_input.h
+++ b/applications/services/gui/modules/text_input.h
@@ -70,6 +70,8 @@ void text_input_set_validator(
     TextInputValidatorCallback callback,
     void* callback_context);
 
+void text_input_set_minimum_length(TextInput* text_input, size_t minimum_length);
+
 TextInputValidatorCallback text_input_get_validator_callback(TextInput* text_input);
 
 void* text_input_get_validator_callback_context(TextInput* text_input);
diff --git a/applications/services/gui/modules/variable_item_list.c b/applications/services/gui/modules/variable_item_list.c
index cf7f64ca3..3cb58b19d 100644
--- a/applications/services/gui/modules/variable_item_list.c
+++ b/applications/services/gui/modules/variable_item_list.c
@@ -26,6 +26,8 @@ typedef struct {
     VariableItemArray_t items;
     uint8_t position;
     uint8_t window_position;
+
+    FuriString* header;
 } VariableItemListModel;
 
 static void variable_item_list_process_up(VariableItemList* variable_item_list);
@@ -125,6 +127,22 @@ uint8_t variable_item_list_get_selected_item_index(VariableItemList* variable_it
     return idx;
 }
 
+void variable_item_list_set_header(VariableItemList* variable_item_list, const char* header) {
+    furi_assert(variable_item_list);
+
+    with_view_model(
+        variable_item_list->view,
+        VariableItemListModel * model,
+        {
+            if(header == NULL) {
+                furi_string_reset(model->header);
+            } else {
+                furi_string_set_str(model->header, header);
+            }
+        },
+        true);
+}
+
 static bool variable_item_list_input_callback(InputEvent* event, void* context) {
     VariableItemList* variable_item_list = context;
     furi_assert(variable_item_list);
@@ -376,6 +394,23 @@ VariableItem* variable_item_list_add(
     return item;
 }
 
+VariableItem* variable_item_list_get(VariableItemList* variable_item_list, uint8_t position) {
+    VariableItem* item = NULL;
+    furi_assert(variable_item_list);
+
+    with_view_model(
+        variable_item_list->view,
+        VariableItemListModel * model,
+        {
+            if(position < VariableItemArray_size(model->items)) {
+                item = VariableItemArray_get(model->items, position);
+            }
+        },
+        true);
+
+    return item;
+}
+
 void variable_item_list_set_enter_callback(
     VariableItemList* variable_item_list,
     VariableItemListEnterCallback callback,
diff --git a/applications/services/gui/modules/variable_item_list.h b/applications/services/gui/modules/variable_item_list.h
index db2a58993..92e5fbedf 100644
--- a/applications/services/gui/modules/variable_item_list.h
+++ b/applications/services/gui/modules/variable_item_list.h
@@ -59,6 +59,15 @@ VariableItem* variable_item_list_add(
     VariableItemChangeCallback change_callback,
     void* context);
 
+/** Get item in VariableItemList
+ *
+ * @param      variable_item_list  VariableItemList instance
+ * @param      position            index of the item to get
+ *
+ * @return     VariableItem* item instance
+ */
+VariableItem* variable_item_list_get(VariableItemList* variable_item_list, uint8_t position);
+
 /** Set enter callback
  *
  * @param      variable_item_list  VariableItemList instance
@@ -74,6 +83,14 @@ void variable_item_list_set_selected_item(VariableItemList* variable_item_list,
 
 uint8_t variable_item_list_get_selected_item_index(VariableItemList* variable_item_list);
 
+/** Set optional header for variable item list
+ * Must be called before adding items OR after adding items but also call set_selected_item() after set_header()
+ *
+ * @param      variable_item_list  VariableItemList instance
+ * @param      header              header to set
+ */
+void variable_item_list_set_header(VariableItemList* variable_item_list, const char* header);
+
 /** Set item current selected index
  *
  * @param      item                 VariableItem* instance
diff --git a/targets/f7/api_symbols.csv b/targets/f7/api_symbols.csv
index f852a69be..7f70d770b 100644
--- a/targets/f7/api_symbols.csv
+++ b/targets/f7/api_symbols.csv
@@ -1,5 +1,5 @@
 entry,status,name,type,params
-Version,+,54.0,,
+Version,+,56.1,,
 Header,+,applications/drivers/subghz/cc1101_ext/cc1101_ext_interconnect.h,,
 Header,+,applications/services/bt/bt_service/bt.h,,
 Header,+,applications/services/cli/cli.h,,
@@ -3198,6 +3198,7 @@ Function,+,text_input_get_validator_callback_context,void*,TextInput*
 Function,+,text_input_get_view,View*,TextInput*
 Function,+,text_input_reset,void,TextInput*
 Function,+,text_input_set_header_text,void,"TextInput*, const char*"
+Function,+,text_input_set_minimum_length,void,"TextInput*, size_t"
 Function,+,text_input_set_result_callback,void,"TextInput*, TextInputCallback, void*, char*, size_t, _Bool"
 Function,+,text_input_set_validator,void,"TextInput*, TextInputValidatorCallback, void*"
 Function,-,tgamma,double,double
@@ -3233,10 +3234,12 @@ Function,+,variable_item_get_current_value_index,uint8_t,VariableItem*
 Function,+,variable_item_list_add,VariableItem*,"VariableItemList*, const char*, uint8_t, VariableItemChangeCallback, void*"
 Function,+,variable_item_list_alloc,VariableItemList*,
 Function,+,variable_item_list_free,void,VariableItemList*
+Function,+,variable_item_list_get,VariableItem*,"VariableItemList*, uint8_t"
 Function,+,variable_item_list_get_selected_item_index,uint8_t,VariableItemList*
 Function,+,variable_item_list_get_view,View*,VariableItemList*
 Function,+,variable_item_list_reset,void,VariableItemList*
 Function,+,variable_item_list_set_enter_callback,void,"VariableItemList*, VariableItemListEnterCallback, void*"
+Function,+,variable_item_list_set_header,void,"VariableItemList*, const char*"
 Function,+,variable_item_list_set_selected_item,void,"VariableItemList*, uint8_t"
 Function,+,variable_item_set_current_value_index,void,"VariableItem*, uint8_t"
 Function,+,variable_item_set_current_value_text,void,"VariableItem*, const char*"

