diff --git a/targets/f7/furi_hal/furi_hal_region.c b/targets/f7/furi_hal/furi_hal_region.c
index 143d5c18c..7372cc2bb 100644
--- a/targets/f7/furi_hal/furi_hal_region.c
+++ b/targets/f7/furi_hal/furi_hal_region.c
@@ -35,19 +35,19 @@ const FuriHalRegion furi_hal_region_us_ca_au = {
     .bands_count = 3,
     .bands = {
         {
-            .start = 304100000,
-            .end = 321950000,
+            .start = 300000000,
+            .end = 348000000,
             .power_limit = 12,
             .duty_cycle = 50,
         },
         {
-            .start = 433050000,
-            .end = 434790000,
+            .start = 387000000,
+            .end = 464000000,
             .power_limit = 12,
             .duty_cycle = 50,
         },
         {
-            .start = 915000000,
+            .start = 779000000,
             .end = 928000000,
             .power_limit = 12,
             .duty_cycle = 50,
@@ -107,15 +107,7 @@ const char* furi_hal_region_get_name() {
 }
 
 bool furi_hal_region_is_frequency_allowed(uint32_t frequency) {
-    if(!furi_hal_region) {
-        return false;
-    }
-
-    const FuriHalRegionBand* band = furi_hal_region_get_band(frequency);
-    if(!band) {
-        return false;
-    }
-
+    furi_hal_region_get_band(frequency);
     return true;
 }
 
